inspirehep/modules/migrator/tasks.py

```diff
@@ -379,7 +347,8 @@ def migrate_and_insert_record(raw_record, skip_files=False):
         ensure_valid_schema(json_record)
 
     try:
-        record = record_insert_or_replace(json_record, skip_files=skip_files)
+        record = InspireRecord.create_or_update(json_record, skip_files=skip_files)
+        record.commit()
```

Comment:
Is there no workaround having to call `commit()` every time outside InspireRecord? Or we want that to push several updates with just one commit, even though I believe you can commit subsequently without issues?

Reply:
Good point! The idea is to move the logic of the ``commit`` outside of ``InspireRecord`` so we can call it explicitly when we need it. 

[Record](https://github.com/inveniosoftware/invenio-records/blob/master/invenio_records/api.py#L137) has exactly the same behaviour, for some reason we didn't respect and we had inconsistencies such as ``delete`` was ``commit``ing but ``create`` was ``commit``ing only if ``skip_files`` was ``True``.